lib/Runtime/Base/ThreadContext.h

```diff
@@ -99,24 +99,28 @@ class InterruptPoller
             void * __frameAddr = nullptr; \
             GET_CURRENT_FRAME_ID(__frameAddr); \
             Js::LeaveScriptObject<stackProbe, leaveForHost, isFPUControlRestoreNeeded> __leaveScriptObject(scriptContext, __frameAddr); \
+            BEGIN_NO_EXCEPTION; \
             AutoReentrancyHandler autoReentrancyHandler(scriptContext->GetThreadContext());
 #define LEAVE_SCRIPT_END_EX(scriptContext) \
             if (scriptContext != nullptr) \
                 {   \
                     scriptContext->GetThreadContext()->DisposeOnLeaveScript(); \
                 }\
+            END_NO_EXCEPTION; \
         }
 
-#define LEAVE_SCRIPT_IF(scriptContext, condition, block) \
-        if (condition) \
+#define LEAVE_SCRIPT_IF_ACTIVE(scriptContext, active, externalCall) \
```

Comment:
Why not just let the macro do the isActive check since it already has a script context (and so we can remove the active parameter)?